src/tests/ftest/container/full_pool_container_create.py

```diff
@@ -86,32 +88,56 @@ def test_no_space_cont_create(self):
             "Pool %s query data: %s\n", self.pool.uuid, self.pool.query_data)
 
         # create a 2nd container now that pool is full
-        try:
-            self.log.info("creating 2nd container")
-            cont2 = DaosContainer(self.context)
-            cont2.create(self.pool.pool.handle)
-            self.log.info("created 2nd container")
+        # try writing to this second container which should
+        # fail after writing a few Kb of data as there should
+        # be some space emptied up due to aggregation and
+        # rdb log compression
+        self.log.info("creating 2nd container")
+        cont2 = DaosContainer(self.context)
+        cont2.create(self.pool.pool.handle)
+        self.log.info("created 2nd container")
+        self.log.info("opening container 2")
+        cont2.open()
+
+        written_data_cont2 = 0
+        self.log.info("writing more objects, write expected to fail "
+                      "after writing a few Kb")
+        for obj_sz in [10, 1]:
+            write_count = 0
+            while True:
+                try:
+                    # write to second container
+                    cont2.write_an_obj(my_str, obj_sz, dkey, akey,
+                                       obj_cls="OC_SX")
+                    write_count += 1
 
-            self.log.info("opening container 2")
-            cont2.open()
+                except DaosApiError as excep:
+                    if not (err in repr(excep) or err2 in repr(excep)):
+                        self.log.error("caught unexpected exception while "
+                                       "writing object: %s", repr(excep))
+                        self.log.info("closing container")
+                        cont2.close()
+                        self.fail("caught unexpected exception while writing "
+                                  "object: {}".format(repr(excep)))
+                    else:
+                        # calculate the data written to second container
+                        # and verify it is under the threshold of 0.1%
+                        # of pool size
+                        written_data_cont2 = (written_data_cont2 +
+                                              (write_count * obj_sz))
```

Comment:
I'm not clear about this calculation. This is the value that we wrote/sent to DAOS, not the actual value written to DAOS. Doesn't the outcome of the if statement on line 130 depend on our expectation or logic and not the DAOS behavior?

Reply:
The thinking behind checking the DAOS behavior is to be able to raise the alarm with enough information if anything goes bad. For example, if there is some bug and we are able to write X amount of data which is greater than the expected behaviors of 0.1% of pool size, we know how much greater it is.
If there is any other solution in your mind feel free to pitch it and we can discuss it.

Reply:
I guess my question is about the [10, 1] obj_sz. How did you determine these? 10 might cause it to go beyond the threshold in the first iteration. For example, if the 0.1% threshold is 54, then hopefully DAOS will raise -1007 before we write the 10 six times. Then hopefully it'll raise -1007 before we write 1 five times. But what if the 0.1% threshold is 4. Then the first iteration will go beyond the threshold before giving DAOS any chance of raising -1007.